rest/route/reliability_test.go

```diff
@@ -402,3 +448,165 @@ func TestRunTestHandler(t *testing.T) {
+					url := getURL(projectID, map[string]interface{}{
+						"tasks":         "aggregation_expression_multiversion_fuzzer",
+						"after_date":    "2019-01-02",
+						"group_by_days": "10",
+					})
+
+					sc := &data.MockConnector{
+						MockTaskReliabilityConnector: data.MockTaskReliabilityConnector{},
+						URL:                          url,
+					}
+					handler := makeGetProjectTaskReliability(sc).(*taskReliabilityHandler)
+					assert.NoError(err)
+
+					// 1 page size of documents will be returned
+					sc.MockTaskReliabilityConnector.SetTaskReliabilityScores("aggregation_expression_multiversion_fuzzer", pageSize)
+					handler.filter = reliability.TaskReliabilityFilter{
+						StatsFilter: stats.StatsFilter{
+							Limit: pageSize,
+						},
+					}
+
+					resp := handler.Run(context.Background())
+
+					assert.NotNil(resp)
+					assert.Equal(http.StatusOK, resp.Status())
+					assert.NotNil(resp.Pages())
+					lastDoc := sc.MockTaskReliabilityConnector.CachedTaskReliability[pageSize-1]
+					assert.Equal(lastDoc.StartAtKey(), resp.Pages().Next.Key)
+				},
+				"More Than One Page": func(ctx context.Context, t *testing.T) {
+					assert := assert.New(t)
+					err := setupTest(t)
+					assert.NoError(err)
+					url := getURL(projectID, map[string]interface{}{
+						"tasks":         "aggregation_expression_multiversion_fuzzer",
+						"after_date":    "2019-01-02",
+						"group_by_days": "10",
+					})
+
+					sc := &data.MockConnector{
+						MockTaskReliabilityConnector: data.MockTaskReliabilityConnector{},
+						URL:                          url,
+					}
+					handler := makeGetProjectTaskReliability(sc).(*taskReliabilityHandler)
+					assert.NoError(err)
+
+					// 2 pages of documents are available.
+					sc.MockTaskReliabilityConnector.SetTaskReliabilityScores("aggregation_expression_multiversion_fuzzer", pageSize*2)
+					handler.filter = reliability.TaskReliabilityFilter{
+						StatsFilter: stats.StatsFilter{
+							Limit: pageSize,
+						},
+					}
+
+					resp := handler.Run(context.Background())
+
+					assert.NotNil(resp)
+					assert.Equal(http.StatusOK, resp.Status())
+					assert.NotNil(resp.Pages())
+					lastDoc := sc.MockTaskReliabilityConnector.CachedTaskReliability[pageSize-1]
+					assert.Equal(lastDoc.StartAtKey(), resp.Pages().Next.Key)
+				},
+				"Invalid Start At": func(ctx context.Context, t *testing.T) {
+					assert := assert.New(t)
+					handler := taskReliabilityHandler{}
+
+					values := url.Values{
+						"tasks":    []string{"aggregation_expression_multiversion_fuzzer"},
+						"start_at": []string{"2.0"},
+					}
+
+					err := handler.parseTaskReliabilityFilter(values)
+					assert.NotNil(err)
+
+					resp := err.(gimlet.ErrorResponse)
+					assert.Equal(http.StatusBadRequest, resp.StatusCode)
+					assert.Equal("Invalid start_at value", resp.Message)
+				},
+			} {
+				t.Run(testName, func(t *testing.T) {
+					withSetupAndTeardown(t, env, func() {
+						testCase(ctx, t)
```

Comment:
this is using the context from the parent rather than the context from the other test (I think), which doesn't matter but might be weird, and would mean that these tests wouldn't respect the timeout.